util_test.go

```diff
@@ -310,3 +317,134 @@ func TestEscapeForCSV(t *testing.T) {
+	pemFile := createTmpFile([]byte("gibberish"))
+	_, err := parsePrivateKeyFromFile(pemFile)
+
+	if err == nil {
+		t.Error("should report error for wrong data in file")
+	}
+}
+
+func TestParsePrivateKeyFromFile(t *testing.T) {
+	generatedKey, _ := rsa.GenerateKey(cr.Reader, 1024)
+	pemKey, _ := x509.MarshalPKCS8PrivateKey(generatedKey)
+	pemData := pem.EncodeToMemory(
+		&pem.Block{
+			Type:  "RSA PRIVATE KEY",
+			Bytes: pemKey,
+		},
+	)
+	keyFile := createTmpFile(pemData)
+	defer os.Remove(keyFile)
+
+	parsedKey, err := parsePrivateKeyFromFile(keyFile)
+	if err != nil {
+		t.Errorf("unable to parse pam file from path: %v, err: %v", keyFile, err)
+	} else if !parsedKey.Equal(generatedKey) {
+		t.Errorf("generated key does not equal to parsed key from file\ngeneratedKey=%v\nparsedKey=%v",
+			generatedKey, parsedKey)
+	}
+}
+
+func createTmpFile(content []byte) string {
+	tempFile, _ := os.CreateTemp("", "exampleKey.pam")
+	tempFile.Write(content)
+	absolutePath := tempFile.Name()
+	return absolutePath
+}
+
+func TestGetFromEnv(t *testing.T) {
+	os.Setenv("SF_TEST", "test")
+	defer os.Unsetenv("SF_TEST")
+	result, err := GetFromEnv("SF_TEST", true)
+
+	if err != nil {
+		t.Error("failed to read SF_TEST environment variable")
+	}
+	if result != "test" {
+		t.Errorf("Incorrect value read for SF_TEST. Expected: test, read %v", result)
+	}
+}
+
+func TestGetFromEnvFailOnMissing(t *testing.T) {
+	_, err := GetFromEnv("SF_TEST_MISSING", true)
+	if err == nil {
+		t.Error("should report error when there is missing env parameter")
+	}
+}
+
+type configParamToValue struct {
+	configParam string
+	value       string
+}
+
+func TestGetConfigFromEnv(t *testing.T) {
+	envMap := map[string]configParamToValue{
+		"SF_TEST_ACCOUNT":     {"Account", "account"},
+		"SF_TEST_USER":        {"User", "user"},
+		"SF_TEST_PASSWORD":    {"Password", "password"},
+		"SF_TEST_ROLE":        {"Role", "role"},
+		"SF_TEST_HOST":        {"Host", "host"},
+		"SF_TEST_PORT":        {"Port", "8080"},
+		"SF_TEST_PROTOCOL":    {"Protocol", "http"},
+		"SF_TEST_WAREHOUSE":   {"Warehouse", "warehouse"},
+		"SF_TEST_DATABASE":    {"Database", "database"},
+		"SF_TEST_REGION":      {"Region", "region"},
+		"SF_TEST_PASSCODE":    {"Passcode", "passcode"},
+		"SF_TEST_SCHEMA":      {"Schema", "schema"},
+		"SF_TEST_APPLICATION": {"Application", "application"},
+	}
+	var properties = make([]ConfigParam, len(envMap))
+	for key, ctv := range envMap {
+		os.Setenv(key, ctv.value)
+		properties = append(properties, ConfigParam{ctv.configParam, key, true})
+	}
+	defer func() {
+		for key := range envMap {
+			os.Unsetenv(key)
+		}
+	}()
+
+	cfg, err := GetConfigFromEnv(properties)
+
+	if err != nil {
+		t.Errorf("unable to parse env variables to Config, err: %v", err)
+	}
+	errors := isConfigCorrect(*cfg, envMap)
+	if errors != "" {
+		t.Errorf(errors)
+	}
+}
+
+func isConfigCorrect(cfg Config, envMap map[string]configParamToValue) string {
```

Comment:
Strange - function prefixed with `is` suggests returning boolean...

Reply:
Modified the function a little bit. It now returns error and the name is `checkConfig`.